src/rules/strictTypePredicatesRule.ts

```diff
@@ -207,7 +208,7 @@ function isFunction(t: ts.Type): boolean {
         return true;
     }
     const symbol = t.getSymbol();
-    return (symbol && symbol.getName()) === "Function";
+    return symbol === undefined ? false : symbol.getName() === "Function";
```

Comment:
nit: a simple boolean expression would be slightly easier to read, `symbol !== undefined && symbol.getName() === "Function"`